### PR TITLE
Update release support data

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
**What this PR does / why we need it**:
- Update release support data

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update release support data
```

### CODE REVIEW

File: docs/book/src/developer/release-support-guidelines.md [hunk 1]
```diff
@@ -34,8 +34,10 @@ standard [backport policy](https://github.com/kubernetes-sigs/cluster-api/blob/m
 
 | Minor Release | API Version | Supported Until                                    |
 |---------------|-------------|----------------------------------------------------|
-| v0.7.x        | **v1beta2** | when v0.9.0 will be released                       |
-| v0.6.x        | **v1beta2** | when v0.8.0 will be released                       |
+| v0.9.x        | **v1beta2** | when v1.1.0 will be released                       |
+| v0.8.x        | **v1beta2** | when v1.0.0 will be released                       |
```
Comment (Amulyam24) [inline docs/book/src/developer/release-support-guidelines.md:38]: Is `v1.0.0` for the next CAPIBM release?
We have marked the next version as `v0.10.0` right?

Comment (Prajyot-Parab) [reply docs/book/src/developer/release-support-guidelines.md]: missed it, fixed.

**Pull request merged**